src/Servers/Kestrel/Core/src/Internal/Http/HttpResponsePipeWriter.cs

```diff
@@ -35,7 +35,7 @@ public override void CancelPendingFlush()
         public override void Complete(Exception exception = null)
         {
             ValidateState();
-            _pipeControl.Complete(exception);
+            _ = _pipeControl.CompleteAsync(exception);
```

Comment:
This task needs to be tracked and awaited

Reply:
Good catch. That was only ever meant to be temporary ofc.